src/dynamics/time_integration.jl

```diff
@@ -365,25 +365,23 @@ function timestep!(
     not_first_timestep && particle_advection!(progn, diagn, model.particle_advection)
 end
 
-"""
-$(TYPEDSIGNATURES)
-Main time loop that that initializes output and feedback, loops over all time steps
-and calls the output and feedback functions."""
-function time_stepping!(
-    progn::PrognosticVariables,         # all prognostic variables
-    diagn::DiagnosticVariables,         # all pre-allocated diagnostic variables
-    model::AbstractModel,               # all model components
-)          
-    
+"""$(TYPEDSIGNATURES)
+Initializes a `simulation`. Scales the variables, initializes
+the output, stores initial conditions, initializes the progress meter feedback,
+callbacks and performs the first two initial time steps to spin up the
+leapfrogging scheme."""
+function initialize!(simulation::AbstractSimulation)
```

Comment:
So the first time steps are included in the `initialize!(simulation)` function, which means that when we start the time stepping with `timestep!(simulation)` the simulation will be at a `time = 2Δt`?

Could this be made optional so that we can start timestepping at `time = 0`? 
I wonder if we could also embed those `first_timesteps!` in the `timestep!` function, something like
```
function timestep!(simulation)
   if iter == 1
       first_timestep!()
   elseif iter == 2 
       second_timestep!()
   else
       timestep!()
   end
end
```
This would make it easier to couple. However, if it is not possible we can work around it by resetting the clock.

Reply:
at `time = Δt` (the first steps are smaller) but yes. I see your point that this would be more consistent. The first time step is then actually two, but that should be fine, conceptually they count as one, also given they step forward dt only. I'll make another PR to change that.